pkg/option/config_test.go

```diff
@@ -609,6 +609,67 @@ func TestCheckIPv6NativeRoutingCIDR(t *testing.T) {
 
 }
 
+func TestCheckIPAMDelegatedPlugin(t *testing.T) {
+	tests := []struct {
+		name      string
+		d         *DaemonConfig
+		expectErr error
+	}{
+		{
+			name: "IPAMDelegatedPlugin with local router IPv4 set and endpoint health checking disabled",
+			d: &DaemonConfig{
+				IPAM:            ipamOption.IPAMDelegatedPlugin,
+				EnableIPv4:      true,
+				LocalRouterIPv4: "169.254.0.0",
+			},
+			expectErr: nil,
+		},
+		{
+			name: "IPAMDelegatedPlugin with local router IPv6 set and endpoint health checking disabled",
+			d: &DaemonConfig{
+				IPAM:            ipamOption.IPAMDelegatedPlugin,
+				EnableIPv6:      true,
+				LocalRouterIPv6: "fe80::1",
+			},
+			expectErr: nil,
+		},
+		{
+			name: "IPAMDelegatedPlugin with health checking enabled",
+			d: &DaemonConfig{
+				IPAM:                         ipamOption.IPAMDelegatedPlugin,
+				EnableHealthChecking:         true,
+				EnableEndpointHealthChecking: true,
+			},
+			expectErr: fmt.Errorf("EnableEndpointHealthChecking must be disabled with IPAMDelegatedPlugin"),
+		},
+		{
+			name: "IPAMDelegatedPlugin without local router IPv4",
+			d: &DaemonConfig{
+				IPAM:       ipamOption.IPAMDelegatedPlugin,
+				EnableIPv4: true,
+			},
+			expectErr: fmt.Errorf("LocalRouterIPv4 must be provided when IPv4 is enabled with IPAMDelegatedPlugin"),
+		},
+		{
+			name: "IPAMDelegatedPlugin without local router IPv6",
+			d: &DaemonConfig{
+				IPAM:       ipamOption.IPAMDelegatedPlugin,
+				EnableIPv6: true,
+			},
+			expectErr: fmt.Errorf("LocalRouterIPv6 must be provided when IPv6 is enabled with IPAMDelegatedPlugin"),
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			err := tt.d.checkIPAMDelegatedPlugin()
+			if !reflect.DeepEqual(tt.expectErr, err) {
+				t.Errorf("expected error %q but got %q", tt.expectErr, err)
+			}
```

Comment:
Checking the actual error for deep equality is not very idiomatic in Go. I'd suggest to just check whether an error is returned or not, i.e. make `expectErr` a `bool` and then change this to something like:

```suggestion
			hasErr := err != nil
			if tt.expectErr && !hasErr {
				t.Errorf("expected error but got none")
			} else if !expectErr && hasErr {
				t.Errorf("expected no error but got %v", err)
			}
```

Reply:
thanks, updated!